frontend/exporter/src/types/mir.rs

```diff
@@ -430,8 +431,12 @@ fn translate_terminator_kind_call<'tcx, S: BaseState<'tcx> + HasMir<'tcx> + HasO
         unreachable!()
     };
 
-    let ty = func.ty(&s.mir().local_decls, s.base().tcx);
-    let fun = if let TyKind::FnDef(def_id, generics) = ty.kind() {
+    let ty = func.ty(&s.mir().local_decls, tcx);
+    let hax_ty: crate::Ty = ty.sinto(s);
+    let TyKind::Arrow(sig) = hax_ty.kind() else {
+        unreachable!("Attempting to call non-function type: {ty:?}")
```

Comment:
Shall we use `supposely_unreachable_fatal!` here instead? (so that we get a spanned failure)